zerver/views/streams.py

```diff
@@ -320,7 +320,7 @@ def add_subscriptions_backend(
             return json_error(_("You can only invite other Zephyr mirroring users to invite-only streams."))
         subscribers = set(principal_to_user_profile(user_profile, principal) for principal in principals)
     else:
-        subscribers = set([user_profile])
+        subscribers = set()
```

Comment:
I am having trouble seeing how this is perceived as a bug by users. Does this have a GitHub issue somewhere? Or maybe we have had a previous discussion about this that I am unaware of? If so, could you please link me to it? If not, creating a private stream and being automatically subscribed to it makes sense (to me at least). Of course, I could be wrong about this! :)

Reply:
Another side effect of this change that I noticed was that, the creator of a private stream has no access to the stream they created, and there is no indication that the stream was created successfully. I think it would be helpful to have a green bar at the top (similar to the red bar from errors) that indicates that the stream was created successfully.

Reply:
After PR #7524 we are allowing current user to whether they want to subscribe to stream or not. 
So now the case is:
* Current user is not always subscribe to private/public stream if current user checkbox is unchecked and `principal.length` is not zero.
* If `principal.length` is zero then current user is forcefully subscribed to private/public stream. like you can see in code that if length is zero we add `user_profile` in `subscribers`.

Reply:
But you are right about having some notification on top of page saying `Stream is successfully created`

Reply:
@YJDave: Cool, that makes sense now that I think about it from the perspective that if the creator's name isn't checked at the time of creation, there is little point in subscribing him to such a stream. Thanks for explaining! :)